06-express-and-layered-architecture/src/cart/cart.controller.ts

```diff
@@ -0,0 +1,74 @@
+import express from "express";
+import * as cartService from "./cart.service";
+import * as orderService from "./order/order.service";
+import joi from "joi";
+import { calculateTotal } from "./middleware/cart.middleware";
+import { checkIfUserExists } from "./middleware/auth.middleware";
+
+const cartUpdateSchema = joi.object({
+  productId: joi.string().required(),
+  count: joi.number().required(),
+});
+
+const router = express.Router();
+router.use("/api/profile/cart", checkIfUserExists);
+
+router.get("/api/profile/cart", async (req, res) => {
+  const userId = res.locals.user;
+  const total = res.locals.total;
+
+  try {
+    const cart = await cartService.getOrCreateCart(userId as string);
+    res.json({ cart, total });
+  } catch (err) {
+    res.status(500).send((err as Error).message);
+  }
+});
+
+router.put("/api/profile/cart", async (req, res) => {
```

Comment:
if you had defined the path in the main entry file, you could have avoided repeating it here three times. Consider defining the router root path in the main entry file in the future.